src/ejabberd_receiver.erl

```diff
@@ -421,3 +423,15 @@ gen_server_call_or_noproc(Pid, Message) ->
     end.
 
 gen_fsm() -> p1_fsm.
+
+hibernate(Result) ->
```

Comment:
Both hibernate/1 and maybe_hibernate/1 needs specs.
I am not fan of element/2 call to get state, because it's not dialyzer verifiable.
Just case can be a better option.